### ISSUE TITLE
web admin: Preview feature doesn't work for User Managers

### ISSUE THREAD

**Issue by nqb** [inverse-inc/packetfence#5619]
**Describe the bug**
When you log as a User Manager and create a user, if you use the Preview button, you got a page without any information.

**To Reproduce**
1. Create a user and assign User Manager access level
2. Log in as user created at step 1
3. Try to create a user
4. Click on preview

=> You got a page without any information.

**Expected behavior**
Display preview like you have if you logged as admin

**Additional context**
After doing several tests, it seems that `CONFIGURATION_MAIN_READ` role is necessary in order to make "Preview" feature work as expected.

User Managers need to have access to:
- `/api/v1/email/preview`

which seems to be dependent to:
- `/api/v1/config/base/alerting`
- `/api/v1/config/base/general`

The problem of giving `CONFIGURATION_MAIN_READ` role to User Managers is that they have a large read access to several configuration items when they only need emails settings.

Related to #5618

**Comment by jrouzierinverse** [comment]
One way to solve this is to redesign our authorization system.
Where a single role to give access to multiple paths and HTTP methods.

For example EMAIL_PREVIEW role give access to GET /api/v1/config/base/alerting, GET /api/v1/config/base/general, and POST /api/v1/email/preview.

This will basically future proof our authorization system as it allows very coarse or fine-grain control of roles to actions.
I am not sure this could be done for the 10.2 release but it could be ready for the release after 10.2

**Comment by julsemaan** [comment]
This should be fairly easy to implement. We'd just need to have an RBAC map that maps a role to additionnal API paths (which would be checked prior the the current map)

So we'd have a new map that looks like this:
```
type apiAccess struct {
  path  string
  verbs []string
}

var dynamicAdminRoleMap = map[string][]apiAccess{
  `EMAIL_PREVIEW`: []apiAccess{
    apiAccess{path: configApiPrefix + `/base/alerting`, verbs: []string{`GET`}},
    apiAccess{path: apiPrefix + `/email/preview`, verbs: []string{`POST`}},
  },
  `NODES_READ`: []apiAccess{
    apiAccess{path: configApiPrefix + `/switches`, verbs: []string{`GET`}},
    apiAccess{path: configApiPrefix + `/switches/search`, verbs: []string{`POST`}},
    apiAccess{path: apiPrefix + `/radius_audit_logs/search`, verbs: []string{`POST`}},
    apiAccess{path: apiPrefix + `/radius_audit_logs`, verbs: []string{`GET`}},
    apiAccess{path: apiPrefix + `/radius_audit_log/`, verbs: []string{`GET`}},
  },
}
```

We should probably have something to handle the search case a bit better so that we don't have to define both the GET on index and POST on search like we have for the existing logic.

**Comment by jrouzierinverse** [comment]
I had something like that in mind but a bit more flexible to deal with sub-routes better.

```go
type pathMatcher interface {
    PathMatched(string) bool
}

type apiAccess struct {
    matcher prefixMatcher
    verbs   []string
}

type prefixMatcher string

func (m prefixMatcher) PathMatched(path string) bool {
    return strings.HasPrefix(path, string(m))
}

var dynamicAdminRoleMap = map[string][]apiAccess{
    `EMAIL_PREVIEW`: []apiAccess{
        apiAccess{matcher: prefixMatcher(configApiPrefix + `/base/alerting`), verbs: []string{`GET`}},
        apiAccess{matcher: prefixMatcher(apiPrefix + `/email/preview`), verbs: []string{`POST`}},
    },
    `NODES_READ`: []apiAccess{
        apiAccess{matcher: prefixMatcher(configApiPrefix + `/switches`), verbs: []string{`GET`}},
        apiAccess{matcher: prefixMatcher(configApiPrefix + `/switches/search`), verbs: []string{`POST`}},
        apiAccess{matcher: prefixMatcher(apiPrefix + `/radius_audit_logs/search`), verbs: []string{`POST`}},
        apiAccess{matcher: prefixMatcher(apiPrefix + `/radius_audit_logs`), verbs: []string{`GET`}},
        apiAccess{matcher: prefixMatcher(apiPrefix + `/radius_audit_log/`), verbs: []string{`GET`}},
    },
}
```

**Comment by julsemaan** [comment]
Yes correct, the code should match all subpaths like it currently does with the existing logic

**Comment by nqb** [comment]
It should work now due to https://github.com/inverse-inc/packetfence/commit/cefecb48b4063321d58106610c1f9c3438dcd574

**Comment by nqb** [comment]
I did a new test on devel. Now we need to have a SYSTEM_CREATE administrative role to be able to use Preview and Send button. Otherwise, GUI doesn't work as expected.

Do you see a reason for that @julsemaan ?

**Comment by julsemaan** [comment]
Let me know on which API call you're getting this, we likely need to adjust the ACL

**Comment by nqb** [comment]
`/api/v1/email/preview` and `/api/v1/email/send`

**Comment by nqb** [comment]
@julsemaan, did you find the time to check ACL ?

I noticed that SYSTEM_CREATE right allow user to access logs using Live logs.

EDIT: Live logs menu is visible from User Manager role by default (without SYSTEM_CREATE right) but they can't create a sesion. If we add SYSTEM_CREATE, they can browse logs.

**Comment by julsemaan** [comment]
The paths authorization roles were just never defined here: https://github.com/inverse-inc/packetfence/blob/devel/go/api-frontend/aaa/authorization.go#L78

They need to be defined and associated an admin role since otherwise it defaults to SYSTEM

**Comment by nqb** [comment]
Thanks for the hint @julsemaan, I will take a look.
